azerothcore/azerothcore-wotlk
# fix(DB/gameobject): Fix overlapping Water Barrel and Food Crate in Kharanos

## Changes Proposed:

This puts both gameobjects into a pool where only one can spawn at a time and gives them an equal spawn chance.

## Issues Addressed:

- Closes #6346
- Closes https://github.com/chromiecraft/chromiecraft/issues/860

## SOURCE:

According to #6346:

> Lack of comments on wowhead/wowpedia remarking an special spawning. Same objects on other zones spawn alone.

## Tests Performed:

See _How to Test the Changes_.

## How to Test the Changes:


1. `.go o 1462`
2. Observe that either the Water Barrel or Food crate is spawned, but not both at the same time
3. Loot the Water Barrel or Food Crate
4. Wait a few minutes for the respawn and observe that either the Water Barrel or Food crate gets spawned, but not both at the same time


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
